src/preact.d.ts

```diff
@@ -39,6 +39,9 @@ declare namespace preact {
 	abstract class Component<PropsType, StateType> implements ComponentLifecycle<PropsType, StateType> {
 		constructor(props?:PropsType);
 
+		static displayName?:string;
+		static defaultProps?:any;
```

Comment:
I had the same thought and even tried it out before this PR. Unfortunately, static members can't reference a class type's parameters. There are ways around this, but I was apprehensive about making the declaration file overly complex.